public/app/percona/dbaas/components/DBCluster/ResourcesBar/ResourcesBar.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { FC } from 'react';
+import { cx } from 'emotion';
+import { Icon, useStyles } from '@grafana/ui';
+import { getStyles } from './ResourcesBar.styles';
+import { ResourcesBarProps } from './ResourcesBar.types';
+import { formatResources, getResourcesWidth } from './ResourcesBar.utils';
+import { Messages } from './ResourcesBar.messages';
+
+export const ResourcesBar: FC<ResourcesBarProps> = ({
+  total,
+  allocated,
+  expected,
+  resourceLabel,
+  units,
+  icon,
+  dataQa,
+  className,
+}) => {
+  const styles = useStyles(getStyles);
+  const requiredResources = allocated && expected ? allocated + expected : undefined;
+  const allocatedWidth = allocated && total ? getResourcesWidth(allocated, total) : 0;
+  const expectedWidth = requiredResources && total ? getResourcesWidth(requiredResources, total) : 0;
```

Comment:
instead of using a ternary every time we call the function, can the function instead check the params and return 0?

Reply:
Of course, good one! Thank yout!